pkg/virt-handler/heartbeat/heartbeat.go

```diff
@@ -174,13 +178,50 @@ func detectCPUManagerFile(cpuManagerPaths []string) (string, error) {
 	return "", fmt.Errorf("no cpumanager policy file found")
 }
 
-func (h *HeartBeat) setSchedulable(labels labelType) {
+func isVirtualizationSupportedOnNodeImpl() bool {
+	// If an error occurs in this process (can't parse info file / any other unexpected failure) this function will
+	// return true just to stay on the safe side. false will be returned only if we are certain that virtualization
+	// is not supported.
+
+	const cpuInfoFilePath = "/proc/cpuinfo"
+
+	cpuInfoFile, err := os.Open(cpuInfoFilePath)
+	if err != nil {
+		log.Log.Reason(err).Errorf("cannot open %s", cpuInfoFilePath)
+		return true
+	}
+	defer cpuInfoFile.Close()
+
+	scanner := bufio.NewScanner(cpuInfoFile)
+	for scanner.Scan() {
+		line := scanner.Bytes()
+		if bytes.Contains(line, []byte("vmx")) || bytes.Contains(line, []byte("svm")) {
+			return true
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Log.Reason(err).Errorf("error while parsing %s", cpuInfoFilePath)
+		return true
+	}
+
+	return false
+}
+
+func (h *HeartBeat) setSchedulable(labels labelType, annotations annotationType) {
 	isKubevirtSchedulable := true
 
 	if !h.deviceManagerController.Initialized() {
 		isKubevirtSchedulable = false
 	}
 
+	if !h.clusterConfig.AllowEmulation() {
```

Comment:
maybe first check `isKubevirtSchedulable` ? no need to check for allowEmulation if its already false right?